Userland/Services/FileOperation/main.cpp

```diff
@@ -64,11 +65,16 @@ struct WorkItem {
     off_t size;
 };
 
+static void report_error(String s)
+{
+    outln("ERROR {} {}", s, strerror(errno));
```

Comment:
It doesn't feel safe to rely on `errno` being intact here, since by this point we've already constructed a `String` object which called `malloc` (and that clobbers `errno`).

Reply:
Ah, right!
Fixed that by caching the errno and building the String in the same way as report_warning does. That should do the trick, shouldn't it?